consensus/propagation/catchup.go

```diff
@@ -108,6 +108,8 @@ func (blockProp *Reactor) AddCommitment(height int64, round int32, psh *types.Pa
 		maxRequests: bits.NewBitArray(int(psh.Total * 2)), // this assumes that the parity parts are the same size
 	}
 
+	blockProp.currentHeight = height + 1
```

Comment:
why is it +1? shouldn't be `height`? since that's the height we're receiving from this commitment

Reply:
when I set it to `height`, weird:
```
2025-05-01 11:38:11 E[2025-05-01|10:38:11.376] Error decoding message                       module=propagation src="Peer{MConn{[fd80:b10c::b]:26656} 30108a336fe6a22a1c78557d600e88c11571499b out}" chId=80 err="invalid Type"
```
 start happening 😆

Reply:
with the new changes in feature/recovery, the issue stopped happening locally.

I guess maybe yes, we should be setting that `blockProp.CurrentHeight` in `addCommimtent` to some value, but since catchup doesn't rely it, it will work. Maybe something else might break due to not setting it, but it's not showing so far

Reply:
also, I've seen a weird case where `blockProp.CurrentHeight` is set to 1017 and the `addCommitment` is adding a height of 1011. This shouldn't happen if I understand correctly

Reply:
also, might be useful to rename `CurrentHeight` to `latestSeenHeight` and `ConsensusHeight` to `latestCommittedHeight` to avoid any confusion for future maintainers

Reply:
the current height is the height that the network is gossiping (or could be in this case). Since we've already received a commitment, we know the network has already moved onto the next height

the consensus height should be = height tho!

Reply:
> with the new changes in feature/recovery, the issue stopped happening locally.

nice! try the modified unit test here on `feature/recovery` for me it fails consistently, but with this change it passes